File: src/main/golang/leetcode/editor/cn/[137]Single Number II.go
package lc

//leetcode submit region begin(Prohibit modification and deletion)
func singleNumber(nums []int) int {
	var ans int32
	for i := 0; i < 32; i++ {
		var total int
		for _, num := range nums {
			total += (num >> i) & 1
		}
		if total%3 > 0 {
			ans |= 1 << i
		}
	}
	return int(ans)
}

//leetcode submit region end(Prohibit modification and deletion)

File: src/main/golang/leetcode/editor/cn/[137]Single Number II.py
from typing import List


# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        ans = 0
        for i in range(32):
            total = sum((num >> i) & 1 for num in nums)
            if total % 3:
                if i==31:
                    ans-= 1 << i
                else:
                    ans |= 1 << i
        return ans

# leetcode submit region end(Prohibit modification and deletion)
